### PR TITLE
Raj_Assignment_19_90min_Done

### PR DESCRIPTION
**Pull request by RajAJun20**
Assignment_19_alldifferString
Assignment_19_AllIndex
Assignment_19_FirstDifferIndex
Assignment_19_Max
Assignment_19_PrimeNumber

### CODE REVIEW

File: src/rajAJun20/AllDifferIndexString.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package rajAJun20;
+
+public class AllDifferIndexString {
+	
+	void allDifferIndexOfStringArray(String[] arr1 , String[] arr2){
+		
+		if(arr1.length == arr2.length) {
+			for(int index = 0 ; index < arr1.length ; index++ ) {
+				if(arr1[index] != arr2[index]) {
```
Comment (KrishnaKTechnocredits) [inline src/rajAJun20/AllDifferIndexString.java]: For string comparison you should always use .equals() method.

File: src/rajAJun20/FirstDifferIndex.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+public class FirstDifferIndex {
+	
+	void differFirstIndexOfArrays(int[] arr1 , int[] arr2 ) {
+		
+		if(arr1.length == arr2.length) {
+				for(int index = 0 ; index < arr1.length ; index++ ) {
+					if(arr1[index] != arr2[index]) {
+						System.out.println("Numbers of both the arrays are not matching at index : " + index);
+					}
```
Comment (KrishnaKTechnocredits) [inline src/rajAJun20/FirstDifferIndex.java:12]: Where is the return statement ??? differFirstIndexOfArrays() we are saying on first index what if other index value is also different.

File: src/rajAJun20/MaximumDifferIndexValue.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package rajAJun20;
+
+public class MaximumDifferIndexValue {
+	
+	void maxValueOfDifferIndex(int[] arr1 , int[] arr2) {
+		if(arr1.length == arr2.length) {
+			for(int index = 0 ; index < arr1.length ; index++) {
+				if(arr1[index] != arr2[index]) {
+					System.out.println("Numbers are not matching at index : " + index);
+				if(arr1[index] > arr2[index])
```
Comment (KrishnaKTechnocredits) [inline src/rajAJun20/MaximumDifferIndexValue.java]: Again this is wrong. Here you should write second if inside first if.

File: src/rajAJun20/PrimeNumberOperations.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+			}
+		}
+		if (flag == true) {
+			System.out.println("Number "+number+ " is not Prime");
+		}
+		else 
+			System.out.println("Number "+ number+ " is Prime");
+	}
+	
+	int totalCountOfPrimeNumbers(){
```
Comment (KrishnaKTechnocredits) [inline src/rajAJun20/PrimeNumberOperations.java:24]: Where is the range ? Method should be parameterized.

File: src/rajAJun20/PrimeNumberOperations.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+				}
+			}
+			if (flag == false) {
+				sum = sum + i;
+			}
+		}
+		return sum;
+	}
+
+	void avgOfPrimeNumbers(){
```
Comment (KrishnaKTechnocredits) [inline src/rajAJun20/PrimeNumberOperations.java:54]: No need to write extra methods, Just create two instance variable sum and count and perform that operation in isPrime method and use those variable in method like:
void findPrimeNumberInGivenRange(int startNumber, int endNumber) {
for(int index=startNumber; index<=endNumber; index++) {
verifyNumberIsPrimeOrNot2(index);
}
System.out.println("Total count is: "+count);
System.out.println("Sum of all Prime number is: "+sum);
System.out.println("Average of all prime number is: "+sum/count);

**Review by KrishnaKTechnocredits — CHANGES_REQUESTED** [review]
.

File: src/rajAJun20/PrimeNumberOperations.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+		}
+		return count;
+	}
+
+	int sumOfPrimeNumbers(){
+		int sum = 0;
+		for (int i = 2; i < 100; i++) {
+			boolean flag = false;
+			for (int j = 2; j < i; j++) {
+				if ( i%j == 0) {
```
Comment (KrishnaKTechnocredits) [inline src/rajAJun20/PrimeNumberOperations.java]: Here you should use numberIsPrimeOrNot() method.

File: src/rajAJun20/PrimeNumberOperations.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+		else 
+			System.out.println("Number "+ number+ " is Prime");
+	}
+	
+	int totalCountOfPrimeNumbers(){
+		int count = 0;
+		for (int i = 2; i < 100; i++) {
+			boolean flag = false;
+			for (int j = 2; j < i ; j++) {
+				if(i % j == 0 ) {
```
Comment (KrishnaKTechnocredits) [inline src/rajAJun20/PrimeNumberOperations.java]: Here you should use numberIsPrimeOrNot() method.

File: src/rajAJun20/AllDifferIndexString.java (hunk 1 above)
Comment (RajAJun20) [reply src/rajAJun20/AllDifferIndexString.java]: Done

File: src/rajAJun20/FirstDifferIndex.java (hunk 2 above)
Comment (RajAJun20) [reply src/rajAJun20/FirstDifferIndex.java:12]: Used break now

File: src/rajAJun20/MaximumDifferIndexValue.java (hunk 3 above)
Comment (RajAJun20) [reply src/rajAJun20/MaximumDifferIndexValue.java]: Done

File: src/rajAJun20/PrimeNumberOperations.java (hunk 7 above)
Comment (RajAJun20) [reply src/rajAJun20/PrimeNumberOperations.java]: Changed method to print number is prime or not and called primeNumber method to show count and sum of prime numbers

**Pull request merged**